clang/include/clang/Driver/Options.td

```diff
@@ -5332,10 +5332,10 @@ def : Flag<["-"], "fno-sycl-explicit-simd">,
 defm sycl_early_optimizations : OptOutCC1FFlag<"sycl-early-optimizations", "Enable", "Disable", " standard optimization pipeline for SYCL device compiler", [CoreOption]>,
   MarshallingInfoFlag<CodeGenOpts<"DisableSYCLEarlyOpts">>;
 def fsycl_dead_args_optimization : Flag<["-"], "fsycl-dead-args-optimization">,
-  Group<sycl_Group>, Flags<[NoArgumentUnused, CoreOption]>, HelpText<"Enables "
+  Group<sycl_Group>, Flags<[CoreOption]>, HelpText<"Enables "
```

Comment:
Yes.  We expect all `-fsycl-*` options  to emit a warning or an error when passed to the driver without `-fsycl` 
The `NoArgumentUnused` flag needs to be removed, otherwise, the warnings gets suppressed.
This one here is for `fsycl_dead_args_optimization`